source/cdk/lib/media-services-application-mapper.ts

```diff
@@ -0,0 +1,156 @@
+/*! Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+       SPDX-License-Identifier: Apache-2.0 */
+
+import {
+    Aws,
+    CfnElement,
+    Fn,
+    Stack,
+    StackProps,
+    CfnParameter,
+} from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import { MsamDynamoDB } from './msam-dynamodb';
+import { MsamIamRoles } from './msam-iam-roles';
+import { MsamEvents } from './msam-events';
+import { MsamBrowserApp } from './msam-browser-app'
+import { MsamCore } from './msam-core';
+import * as utils from './utils';
+
+export class MediaServicesApplicationMapper extends Stack {
+    constructor(scope: Construct, id: string, props?: StackProps) {
+        super(
+            scope,
+            id,
+            {
+                ...props,
+                description: '(SO0048) Media Services Application Mapper Root Stack %%VERSION%%',
+            }
+        );
+
+        /**
+         * Parameters
+         */
+        const cacheItemTTL = new CfnParameter(this, 'CacheItemTTL', {
+            default: '86400',
+            description: 'This is the maximum time in seconds a cached item will remain if never updated (3600 = 1 hour, 86400 = 1 day, 604800 = 1 week).',
+            type: 'String',
+            allowedPattern: '\\S+',
+            minLength: 1,
+            constraintDescription: 'Please enter a value for this field.',
+        });
+
+        /**
+         * Nested Stacks
+         */
+
+        // IAM Module Stack
+        const iamModuleStack = new MsamIamRoles(this, 'IAMModuleStack');
+        utils.addTemplateUrl(
+            iamModuleStack,
+            Fn.join('', [
+                'https://%%BUCKET_NAME%%-',
+                Aws.REGION,
+                '.s3.amazonaws.com/%%SOLUTION_NAME%%/%%VERSION%%/msam-iam-roles-release.template',
```

Comment:
My primary goal was really to match the CFN template as closely as possible, so I used the `Fn.Join` in the CDK port as well. Otherwise, every single `Fn.Join` can really be replaced with the template literals or with built in JS string array join function. I did however clean up the nested `Fn.Join` which I found to be extra redundant.